### DISCUSSION TITLE
run torch 2.5.1 + xformers-0.0.28.post3

### DISCUSSION THREAD

**Discussion by DmitryKonit** [lllyasviel/stable-diffusion-webui-forge#2439] [Show and tell]
Python 3.11.10
![image](https://github.com/user-attachments/assets/ced5c89a-479b-4de9-bc53-9561fb53749f)

**Comment by Juqowel** [comment]
What exactly do you want to say?

**Comment by DmitryKonit** [comment]
[Forge with CUDA 12.4 + Pytorch 2.4](https://github.com/lllyasviel/stable-diffusion-webui-forge/releases/download/latest/webui_forge_cu124_torch24.7z) <- Fastest, but MSVC may be broken, xformers may not work
xformers not work in Pytorch 2.4, in Pytorch 2.5.1 it work

**Comment by yeoldecoot** [comment]
How did you get xformers working with torch 2.5.1+cu124? whenever I try I get an error in the subprocess saying `ModuleNotFoundError: No module named 'torch'`

**Reply by MisterChief95** [reply]
@yeoldecoot try this (make sure venv is activated):
```
path/to/venv/python.exe -m pip install "xformers==0.0.29.post1" --index-url https://download.pytorch.org/whl/cu124
```
The prebuilt wheels for CUDA 12.1/12.4 require the `--index-url` option. If you forget it, it will try to build xformers from source code.

**Reply by yeoldecoot** [reply]
Thank you for the reply. I did eventually figure this out but I forgot to mention it here. It's too bad though that xformers did very little to my generation times on my 3060.
